File: example/builtin/test_task.py
import xuanwu

# 内建工具任务管理与执行控制测试示例 (Python)
# 
# 前提条件：
# 1. 已经在环境中安装了 xuanwu 包（例如通过 `xuanwu builtininstall`）
# 2. 环境中已注入有效环境变量：
#    - XWPKG_OPENAPI_TOKEN (必填，用于管理接口鉴权)

def main():
    print("====== 开始运行 Python 任务管理与执行控制示例 ======")

    try:
        # 获取所有任务列表
        tasks = xuanwu.get_tasks()
        print(f"成功获取到 {len(tasks)} 个定时任务:")
        for task in tasks[:5]:  # 仅打印前5个
            print(f"  - [{task.get('id')}] {task.get('name')} (表达式: {task.get('schedule')}, 备注: {task.get('remark')})")

        # 示例：尝试触发第一个任务的运行
        if tasks:
            target_task = tasks[0]
            print(f"\n尝试手动触发任务运行: [{target_task.get('id')}] {target_task.get('name')}...")
            result = xuanwu.execute_task(target_task.get("id"))
            print("执行指令发送成功。")

        # 获取最近的执行结果列表
        results = xuanwu.get_last_results()
        print(f"\n最近共有 {len(results)} 条任务执行记录。")
            
    except Exception as e:
        print(f"任务操作失败: {e}")
        print("提示: 请确保在面板任务设置中正确注入了 OpenAPI Token。")

if __name__ == "__main__":
    main()

File: example/builtin/test_task.js
const xuanwu = require('xuanwu');

/**
 * 内建工具任务管理测试示例 (Node.js)
 * 
 * 使用说明：
 * 1. 确保已在该 Node.js 环境下安装过内建包。
 * 2. 运行时系统需要注入有效的 OpenAPI 凭证 (XWPKG_OPENAPI_TOKEN 或 OPENAPI_TOKEN)。
 */

async function main() {
    console.log("====== 开始运行 Node.js 任务管理与执行控制示例 ======");

    try {
        // 获取所有任务列表
        const tasks = await xuanwu.getTasks();
        console.log(`成功获取到 ${tasks.length} 个定时任务:`);
        tasks.slice(0, 5).forEach(task => { // 仅展示前5项
            console.log(`  - [${task.id}] ${task.name} (表达式: ${task.schedule || ''}, 备注: ${task.remark || ''})`);
        });

        // 尝试触发第一个任务的运行
        if (tasks.length > 0) {
            const targetTask = tasks[0];
            console.log(`\n尝试手动触发任务运行: [${targetTask.id}] ${targetTask.name}...`);
            await xuanwu.executeTask(targetTask.id);
            console.log("执行指令发送成功。");
        }

        // 获取最近的执行结果列表
        const results = await xuanwu.getLastResults();
        console.log(`\n最近共有 ${results.length} 条任务执行记录。`);

    } catch (e) {
        console.error(`任务操作失败: ${e.message}`);
        console.log("提示: 请确保在面板任务设置中正确注入了 OpenAPI Token。");
    }
}

main();
